#  Baseline Modeling : Hypertension Risk Atlas
**ADS-599 Capstone Project · Group 2 · Nancy Walker & Michelle Wang**

---

This notebook performs baseline modeling for the Hypertension Risk Atlas project. The goal is to establish a foundational understanding of the data and to create initial predictive models for hypertension risk.

## 1. Import Libraries

In [9]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from pathlib import Path
from sklearn.pipeline import Pipeline
from sklearn.preprocessing import StandardScaler
from sklearn.feature_selection import SelectFromModel
from sklearn.linear_model import Lasso, LinearRegression
from sklearn.decomposition import PCA
from sklearn.model_selection import cross_val_score, cross_val_predict
from sklearn.metrics import mean_absolute_error, median_absolute_error

## 2. Load Datasets 

In [10]:
# Path setup
base_path = Path.cwd().parents[1]
processed_dir = base_path / "data" / "final" 

# Ensure the folder exists (best practice)
processed_dir.mkdir(parents=True, exist_ok=True)

# Load the files
X_train = pd.read_csv(processed_dir / "X_train.csv", index_col=0)
X_test = pd.read_csv(processed_dir / "X_test.csv", index_col=0)
y_train = pd.read_csv(processed_dir / "y_train.csv", index_col=0)
y_test = pd.read_csv(processed_dir / "y_test.csv", index_col=0)

# Quick verification
print(f"Train shape: {X_train.shape}")
print(f"Test shape: {X_test.shape}")

Train shape: (1832, 43)
Test shape: (455, 43)


In [11]:
# List of columns to drop (identifiers)
drop_cols = ['fipscode', 'locationname', 'stateabbr', 'State', 'County']

# Drop them from both X_train and X_test
X_train_model = X_train.drop(columns=drop_cols)
X_test_model = X_test.drop(columns=drop_cols)

## 3. Baseline: Linear Regression

Linear Regression modeling will use the robust pipeline (StandardScaler $\rightarrow$ LASSO [$\alpha=0.0055$] $\rightarrow$ PCA [$n=15$]). Modeling will be performed in pipelines to prevent data leakage and ensure reproducibility.

In [12]:
# Define Pipeline 
baseline_pipeline = Pipeline([
    ('scaler', StandardScaler()), 
    ('feature_selection', SelectFromModel(Lasso(alpha=0.0055, max_iter=10000))), 
    ('pca', PCA(n_components=15, random_state=42)),      
    ('regressor', LinearRegression()) # The only difference from your advanced model!
])

## 4. Evaluation

In [13]:
# Evaluate using cross-validation on the training set
baseline_scores = cross_val_score(
    baseline_pipeline, 
    X_train_model, 
    y_train.values.ravel(), 
    cv=5, 
    scoring='r2'
)

baseline_mean_r2 = np.mean(baseline_scores)
print(f"Baseline (Linear Regression) Cross-Validated R²: {baseline_mean_r2:.4f}")

Baseline (Linear Regression) Cross-Validated R²: 0.8437


An $R^2$ of 0.8437 with a simple, linear, and highly interpretable model provides strong evidence that the dataset has a clear, linear signal and the pipeline is successfully capturing.

**Conclusion:** The baseline modeling results indicate that the Hypertension Risk Atlas dataset is suitable for predictive modeling, and the linear regression approach serves as a strong starting point for further model development and refinement. This concludes the Feature Engineering section of the project. The next steps will involve exploring complex modeling techniques and hyperparameter tuning to improve predictive performance and interpretability.